library/Notifications/Widget/Detail/IncidentDetail.php

```diff
@@ -93,28 +90,22 @@ protected function createRelatedObject()
 
     protected function createHistory()
     {
-        $history = $this->incident->incident_history
-            ->with([
-                'event',
-                'incident.object',
-                'incident.object.source',
-                'contact',
-                'rule',
-                'rule_escalation',
-                'contactgroup',
-                'schedule',
-                'channel'
-            ]);
-
-        $history
-            ->withColumns('incident.object.id_tags')
-            ->on(Query::ON_SELECT_ASSEMBLED, function (Select $select) use ($history) {
-                Database::registerGroupBy($history, $select);
-            });
-
         return [
             Html::tag('h2', t('Incident History')),
-            new IncidentHistoryList($history)
+            new IncidentHistoryList(
+                $this->incident->incident_history
+                    ->with([
+                        'incident',
```

Comment:
If you didn't change this, I wouldn't feel the need to comment :sweat_smile: 

This is redundant. The `incident` relation itself is of no use, so remove this join altogether.